clang/test/SemaHLSL/resource_binding_attr_error_mismatch.hlsl

```diff
@@ -0,0 +1,73 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
+
+
+// expected-error@+1  {{invalid register name prefix 'b' for register resource type 'RWBuffer' (expected 'u')}}
+RWBuffer<int> a : register(b2, space1);

```

Comment:
> float b : register(u0, space1);

This would also be invalid even if the register type were correct because you can't have spaces on global constants.